packaging/scripts/postinstall.sh

```diff
@@ -6,6 +6,67 @@ ETC_SERVICE="/etc/systemd/system/evcc.service"
 USR_LOCAL_BIN="/usr/local/bin/evcc"
 RESTART_FLAG_FILE="/tmp/.restartEvccOnUpgrade"
 
+
+# Call /usr/bin/evcc checkconfig and capture the output
+# if exit code is 0, then remove /tmp/oldevcc
+# if exit code is not 0, then fail installation with error message and copy /tmp/oldevcc back to /etc/evcc
+# if /tmp/oldevcc does not exist, then do nothing
+failInstallation=0
+INTERACTIVE=0
+# is shell script interactive?
+if [ -t 0 ]; then
+  INTERACTIVE=1
+else
+  INTERACTIVE=0
+fi
+
+if [ -f /tmp/oldevcc ] && [ $INTERACTIVE -eq 1 ]; then
+    checkConfigOutput=$(/usr/bin/evcc checkconfig 2>&1 || true)
+	oldevccversion=$(/tmp/oldevcc -v)
+	newevccversion=$(/usr/bin/evcc -v)
+    if echo "$checkConfigOutput" | grep -q "config valid"; then
+ 			
+			if [ "$oldevccversion" = "$newevccversion" ]; then
+				echo "--------------------------------------------------------------------------------"
+				echo "Old evcc version detected. To apply the new version, please reinstall evcc. (e.g. apt-get reinstall evcc)"
```

Comment:
This is somewhat confusing. 
The check is for old and new version being equal, yet, the log message claims and old version was detected.
What does "old version" in this context mean?

Reply:
Now I'm even more confused. 😅 
Does the above dpkg output indicate, that the rollback does not actually roll back, in the sense that while it did restore the old version (0.123.8), the package manager now thinks 0.123.9 is already installed?
If so, that sounds like a total nightmare/deal breaker. 😢